src/App.js

```diff
@@ -52,16 +53,38 @@ class PanelButtons extends Component {
     }
   };
 
-  localizedSaveMessage = (saveStatus) => {
-    return I18n.t(`saveStatus_${saveStatus}`)
+  localizedSaveMessage = saveStatus => {
+    return I18n.t(`saveStatus_${saveStatus}`);
+  };
+
+  saveResponseDataMessage = saveResponseData => {
+    // The list of known error types from share_filtering.rb.
+    const errorTypes = ["email", "address", "phone", "profanity"];
+
+    if (!saveResponseData) {
+      return "";
+    }
+
+    const index = errorTypes.indexOf(saveResponseData.type);
+    if (index !== -1) {
+      return `(${index})`;
```

Comment:
Why use an error code rather than telling them the type of error?

Reply:
My idea was that we would start with an incremental change that would let us continue to handle these via support ticket, but we can ask for this number to begin diagnosis.  It seems relatively rare, and will help us understand the nature of problems occurring in the wild.  We can also keep an eye on the new Firehose events to get the relevant data on our end, support incident or not.  

As time goes on, we can certainly add more actionable feedback, but we'll need to figure out the right user experience for that.